clients/drcachesim/tests/memref_gen.h

```diff
@@ -84,7 +84,30 @@ gen_branch_encoded(memref_tid_t tid, addr_t pc, int encoding)
     memref.instr.encoding_is_new = true;
     return memref;
 }
+
+inline memref_t
+gen_syscall_encoded(memref_tid_t tid, addr_t pc)
+{
+    constexpr int encoding = 0xd4000001;
+    memref_t memref = gen_instr_type(TRACE_TYPE_INSTR, tid, pc);
+    memref.instr.size = 4;
+    memcpy(memref.instr.encoding, &encoding, sizeof(encoding));
+    memref.instr.encoding_is_new = true;
+    return memref;
+}
+
 #elif defined(X86_64) || defined(X86_32)
+inline memref_t
+gen_syscall_encoded(memref_tid_t tid, addr_t pc)
```

Comment:
See unresolved comment about not wanting to add a gen_ routine specific to syscalls.